redbox-core/redbox/graph/root.py

```diff
@@ -198,10 +198,10 @@ def get_agentic_search_graph(tools: dict[str, StructuredTool], debug: bool = Fal
     )
     builder.add_edge("p_stuff_docs_agent", "p_report_sources")
     builder.add_edge("p_give_up_agent", "p_report_sources")
+    builder.add_edge("p_stuff_docs_agent", END)
+    builder.add_edge("p_give_up_agent", END)
```

Comment:
Out of interest, why do we need these new endings? Is it for cases where document search takes too long?

Reply:
Explicitly pointing to END is usually for readability of both the code but more importantly the visualisation. Every node sort of implicitly goes to END anyway